lib/pbench/server/api/__init__.py

```diff
@@ -85,18 +82,13 @@ def register_endpoints(api, app, config):
         IndexSearch, f"{base_uri}/index/search", resource_class_args=(config, logger),
     )
     api.add_resource(
-        IterationSampleNamespace,
-        f"{base_uri}/dataset/samples/namespace",
-        resource_class_args=(config, logger),
-    )
-    api.add_resource(
-        IterationSamplesRows,
-        f"{base_uri}/dataset/samples/rows",
+        SampleNamespace,
+        f"{base_uri}/index/namespace/<string:target_document>",
         resource_class_args=(config, logger),
     )
     api.add_resource(
-        TimeSeriesSamples,
-        f"{base_uri}/dataset/samples/timeseries",
+        SampleRows,
+        f"{base_uri}/index/rows/<string:target_document>",
```

Comment:
I think we want this API, at least, to remain under the `/dataset` resource hierarchy. Arguably, though, I think I'm realizing that your `SampleNamespace` should be, too, since while you're returning keywords based on the document mappings, the *values* for each are specific to a dataset. However, I'm having a bit of trouble right now coming up with a good naming hierarchy here.

I do also think that the parameter name `target_document` is misleading: that implies that the dataset name (or run_id) is in the URI. And maybe it *should be* (in a pure REST model), but *this* parameter is really specifying a particular view of the dataset, which is really the granularity of the view (either `result-data-sample` or `result-data`, which, in retrospect, are not particularly well named indices 😆 )

I don't want to tie this to "index", though. Maybe `<string:view>`??

Reply:
Yeah I will add the `dataset` back in the URI, and `<string:view>` sounds okay but its not really a view its a particular index that we will be searching isn't it? So maybe `<string:type>` (for the type of data). Idk, view is also fine.

Reply:
Yeah, "view" doesn't resonate with me, either.  It really is a "data object type" (e.g., result data, or timeseries data), so I would be inclined toward "type", if that's not a too-overloaded term.

```
/dataset/values/<string:type>
```
?

Reply:
Our representation of a dataset is spread across multiple Elasticsearch indices for our own internal convenience; but really it's all one dataset, and we're merely offering various "slices" of that dataset as a summary (run), detailed list of captured files (toc), iteration summaries, or low-level timeseries details. In a REST model I have been inclined to consider representing this all as a `GET /api/v2/dataset/<name>?view=timeseries` ... which was what inspired the "view" name here.

We definitely don't want "index" because we want to move away from directly passing on the Elasticsearch model; "type" seems a bit too generic. The fact that this particular aspect of the dataset is represented in a distinct Elasticsearch "document type" (and index) is an implementation detail we'd rather not expose. The REST resource type is "dataset" ... we're just presenting a particular aspect of the dataset here. I'm not tied to "view"  in this context, and it doesn't really fit the original inspiration of using the semi-standard "view" query parameter; but I struggled to find a better term for a moment and then gave up.

We could consider it "level of detail", but that wouldn't be a clear representation, either. On the whole, I think I prefer "view" over "type", or "index" and right now I don't have a better suggestion. But, fine, you guys like "type"; in the context of the whole v1 API, which is a fairly ugly first-pass expedient hack based on the old dashboard rather than a consistent or meaningful REST model, I'll go along with the majority here.